src/SUMMARY.md

```diff
@@ -14,7 +14,7 @@
     - [Bootstrapping](./building/bootstrapping.md)
     - [Distribution artifacts](./building/build-install-distribution-artifacts.md)
     - [Documenting Compiler](./building/compiler-documenting.md)
-    - [Rustdoc](./rustdoc.md)
+    - [Rustdoc](./building/rustdoc.md)
```

Comment:
So, the problem here is that "The Rustc Driver and Interface" has also the rustdoc chapter, not the path.
It's introduced by 093b9032de1821f8a9f507f80f985aca1920858b but I assume it isn't intentional (cc @mark-i-m). I think it's a better way just to remove that entry, not to change the path.